app/lib/submission_builder/ty2022/states/authentication_header.rb

```diff
@@ -16,8 +16,40 @@ def document
                 xml.NoFinancialProduct "X"
               end
             end
-            xml.PrimDrvrLcnsOrStateIssdIdGrp do
-              xml.DoNotHaveDrvrLcnsOrStIssdId "X"
+            if @submission.data_source.state_name == "New York"
+              xml.PrimDrvrLcnsOrStateIssdIdGrp do
+                state_id = @submission.data_source.primary_state_id
+                if state_id.present? && (state_id.id_type_driver_license? || state_id.id_type_dmv_bmv?)
```

Comment:
it looks like `state_id` is "not null" in the database so probably you don't need the `state_id.present?` check